shell/browser/ui/file_dialog_win.cc

```diff
@@ -144,10 +144,14 @@ static void ApplySettings(IFileDialog* dialog, const DialogSettings& settings) {
   // We set file extension to the first none-wildcard extension to make
   // sure the dialog will update file extension automatically.
   for (size_t i = 0; i < filterspec.size(); ++i) {
-    if (std::wstring(filterspec[i].pszSpec) != L"*.*") {
+    std::wstring spec(filterspec[i].pszSpec);
+    // only need first extension
+    std::wstring firstSpec = spec.substr(0, spec.find(L';'));
+    if (firstSpec != L"*.*") {
```

Comment:
In cases where `spec` is `*.*`, I think the substr will not be necessary.
```suggestion
    std::wstring spec(filterspec[i].pszSpec);
    if (spec != L"*.*") {
```